django/db/backends/oracle/operations.py

```diff
@@ -452,10 +454,17 @@ def _get_trigger_name(self, table):
         return '%s_TR' % truncate_name(table, name_length).upper()
 
     def bulk_insert_sql(self, fields, placeholder_rows):
-        return " UNION ALL ".join(
-            "SELECT %s FROM DUAL" % ", ".join(row)
-            for row in placeholder_rows
-        )
+        query = list()
+        for row in placeholder_rows:
+            select = list()
+            for i, ph in enumerate(row):
+                if not fields[i]:
```

Comment:
Add a comment explaining under what circumstances the two branches are applicable?